dames.c

```diff
@@ -71,66 +71,164 @@ char pion_init(int ligne, int colonne){
     return pion;
 }
 
+int changement_dans_damier(char tab[][COLONNE_DAMIER],char tab2[][COLONNE_DAMIER]){
 
-
+    for (int ligne = 0; ligne < LIGNE_DAMIER ; ligne++){
+        for(int colonne = 0 ; colonne < COLONNE_DAMIER; colonne++){
+            if(tab2[colonne][ligne]!=tab[colonne][ligne]){
+                return 1;
+            }
+        }
+    }
+    
+return 0;
+}
 
 
 //DEBUT PARTIE
 void jouer_dames(char tab[][COLONNE_DAMIER], char couleur){
 
-    char partie[20]= "start";
+    char tab2[COLONNE_DAMIER][LIGNE_DAMIER];
+
+    char partie= 'S';
     printf("Les BLANC commencent \n");
-    while(partie != "terminer")
-    {
+    while(partie != 'T'){
+        do{
+            for (int ligne = 0; ligne < LIGNE_DAMIER ; ligne++){
+                for(int colonne = 0 ; colonne < COLONNE_DAMIER; colonne++){
+                    tab2[colonne][ligne]=tab[colonne][ligne];
+                }
+            }
+            action_automatique(tab,couleur);
+
+            changement_dans_damier(tab,tab2);
+
+            if (changement_dans_damier(tab,tab2)==1){
+                if (couleur == 'B'){
+                couleur = 'W';
+                }
+                else{
+                couleur = 'B';  
+                }
+            }
+    
+        }while(changement_dans_damier(tab,tab2) == 1);
+    
     Choix_du_Pion(tab,couleur);
-    if (couleur == 'B'){
-        couleur = 'W';
-    }
-    else{
-        couleur = 'B';  
+
+        if (couleur == 'B'){
+            couleur = 'W';
+        }
+        else{
+            couleur = 'B';  
+        }
+
+    printf("pion %c, à vous de jouer !\n",couleur);
     }
     
-    printf("pion %c, à vous de jouer !\n",couleur);
-
     }
 
-}
+void action_automatique(char tab[][COLONNE_DAMIER], char couleur){
 
-Void action_automatique(char tab[][COLONNE_DAMIER], char couleur){
+    int tab_ligne_pion_qui_peut_manger[100];
+    int tab_colonne_pion_qui_peut_manger[100];
```

Comment:
What 100 refers to? Simple guess: Can't it be 20 the maximum number of pawn per player or 10 the maximum number of line/column ?